ArduPlane/servos.cpp

```diff
@@ -551,6 +553,15 @@ float Plane::apply_throttle_limits(float throttle_in)
         min_throttle = 0;
     }
 
+    // Handle throttle limits for transition conditions.
+#if HAL_QUADPLANE_ENABLED
+    if (quadplane.in_transition()) {
```

Comment:
why not this?
![image](https://github.com/user-attachments/assets/d629479a-0e16-443b-8c74-8bbf3be24e7d)

Reply:
note that in_transition() is only true for fwd transitions

Reply:
turns out in_transiton() is true for tailsitters all the time, which doesn't seem right
I think we need a new in_forward_transition() which is only true for the fwd transition